Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -370,34 +370,104 @@ class Gx2Fitter {
           extensions.calibrator(state.geoContext, *calibrationContext,
                                 sourcelink_it->second, trackStateProxy);
 
-          const size_t measdimPlaceholder = 2;
-          auto measurement =
-              trackStateProxy.template calibrated<measdimPlaceholder>();
-          auto covarianceMeasurement =
-              trackStateProxy
-                  .template calibratedCovariance<measdimPlaceholder>();
-          // calculate residuals and return with covariances and jacobians
-          ActsVector<2> residual;
-          for (long i = 0; i < measurement.size(); i++) {
-            residual[i] = measurement[i] - predicted[i];
+          if (trackStateProxy.calibratedSize() == 1) {
```

Comment:
Do you think you could do with unrolling ?

Something like 
```c++
template <size_t kDIM>
processMeasurement<kDIM>() {... };
```

and then unroll via a `std::integer_sequence` ?

Reply:
I think this could be for n-dimensional measurements done with the `visit_measurement` function. Or do you think unrolling is necessary here?